tutorials/video_classification_example/finetune.py

```diff
@@ -0,0 +1,255 @@
+        self.num_classes = len(self.classes)
+
+
+        # TODO - too many repeated .glob calls here.
+        self.train_paths = []
+        self.val_paths = []
+        self.holdout_scenes = {}
+        for c in self.classes:
+
+            # Scenes within each class directory
+            scene_names = sorted(set(x.name for x in (self.root / c).glob("*") if x.is_dir() and x.name != 'Annotation'))
+            
+            # Holdout the last scene
+            # TODO - wrap this in a function so users can override the split logic
+            holdout_scene = scene_names[-1]
+            scene_names = scene_names[:-1]
+
+            # Keep track of which scenes we held out for each class w/ a dict
+            self.holdout_scenes[c] = holdout_scene
+
+            # Prepare the list of 'labeled paths' required by the LabeledVideoDataset
+            label_paths = [(v, {"label": self.class_to_label[c]}) for v in (self.root / c).glob("**/*.avi")]
+
+            # HACK - this is no bueno. Can be done within the loop above
+            self.train_paths.extend([x for x in label_paths if x[0].parent.name != holdout_scene])
+            self.val_paths.extend([x for x in label_paths if x[0].parent.name == holdout_scene])
+
+    def _video_transform(self, mode: str):
+        # TODO - different tsfm for val/train
+        return ApplyTransformToKey(
+            key="video",
+            transform=Compose(
+                [
+                    UniformTemporalSubsample(self.num_frames),
+                    Lambda(lambda x: x / 255.0),
+                    Normalize(self.mean, self.std),
+                    ShortSideScale(size=self.side_size),
+                    CenterCropVideo(crop_size=(self.crop_size, self.crop_size)),
+                ]
+            ),
+        )
+
+    def _make_dataset(self, mode: str):
+        """
+        Defines the train DataLoader that the PyTorch Lightning Trainer trains/tests with.
+        """
+        sampler = DistributedSampler if (self.trainer is not None and self.trainer.use_ddp) else RandomSampler
+        return LimitDataset(LabeledVideoDataset(
+            self.train_paths if mode == 'train' else self.val_paths,
+            UniformClipSampler(self.clip_duration),
+            decode_audio=False,
+            transform=self._make_transforms(mode=mode),
+            video_sampler=sampler,
+        ))
+
+    def train_dataloader(self):
+        self.train_dataset = self._make_dataset('train')
+        return torch.utils.data.DataLoader(
+            self.train_dataset,
+            batch_size=self.args.batch_size,
+            num_workers=self.args.workers,
+        )
+
+    def val_dataloader(self):
+        self.val_dataset = self._make_dataset('val')
+        return torch.utils.data.DataLoader(
+            self.val_dataset,
+            batch_size=self.args.batch_size,
+            num_workers=self.args.workers,
+        )
+
+
+class MiniKineticsDataModule(KineticsDataModule):
```

Comment:
Rather than creating an inheritance hierarchy can you just make these arguments to the original kinetics data module class?